_aggregations/metric/scripted-metric.md

```diff
@@ -55,18 +231,39 @@ GET opensearch_dashboards_sample_data_logs/_search
 ```
 {% include copy-curl.html %}
 
-#### Example response
+Example response
+
+The response returns three values together under the `value` object. This demonstrates how a scripted metric can return multiple metrics at once by using a map in the state:
 
 ```json
-...
-"aggregations" : {
-  "responses.counts" : {
-    "value" : {
-      "other" : 0,
-      "success" : 12832,
-      "error" : 1242
+{
+  ...
+  "hits": {
+    "total": {
+      "value": 5,
+      "relation": "eq"
+    },
+    "max_score": null,
+    "hits": []
+  },
+  "aggregations": {
+    "responses_by_type": {
+      "value": {
+        "other": 1,
+        "success": 2,
+        "error": 2
+      }
     }
   }
- }
 }
 ```
+
+## Handling empty buckets (no documents scenario)
+
+When using a `scripted_metric` aggregation as a sub-aggregation within a bucket aggregation (such as terms), it is important to account for buckets that contain no documents on certain shards. In such cases, those shards return a `null` value for the aggregation state. During the `reduce_script` phase, the states array may therefore include `null` entries corresponding to these shards. To ensure reliable execution, the `reduce_script` must be designed to handle `null` values gracefully. A common approach is to include a conditional check, such as `if (state != null)`, before accessing or operating on each state. Failure to implement such checks can result in runtime errors when processing empty buckets across shards.
```

Comment:
Not related to this documentation - but do you think that handling null can be another improvement in code base by passing some paramater like `ignore_null_results`.


I'm wondering that might even make scripted aggs faster if the null checks are part of code-flow via some param rather than run as part of script.